include/slam_toolbox/slam_toolbox_common.hpp

```diff
@@ -172,6 +183,7 @@ class SlamToolbox : public rclcpp::Node
   ProcessType processor_type_;
   std::unique_ptr<karto::Pose2> process_near_pose_;
   tf2::Transform reprocessing_transform_;
+  bool do_threads_clean_up_;
```

Comment:
Remove, just check if the thread is valid and `joinable()` for joining them. No need to maintain state on it

Reply:
https://github.com/SteveMacenski/slam_toolbox/pull/561/commits/7f89be2cbcb0cf77349f8e05d58887e1d126983b

Removed `do_threads_clean_up_`. Instead, threads are now stopped with `get_current_state`.